src/foundry/foundry.js/applications/formApplications/documentSheet.d.ts

```diff
@@ -1,106 +1,123 @@
+     * ```
+     */
+    static get defaultOptions(): DocumentSheet.Options;
 
-  /**
-   * @deprecated since 0.8.0
-   */
-  get entity(): O;
+    /**
+     * A semantic convenience reference to the Document instance which is the target object for this form.
+     */
+    get document(): O;
 
-  /**
-   * @override
-   */
-  get isEditable(): boolean;
+    /**
+     * @override
+     */
+    get id(): string;
 
-  /**
-   * @override
-   */
-  render(force?: boolean, options?: Application.RenderOptions): this;
+    /**
+     * @override
+     */
+    get isEditable(): boolean;
 
-  /**
-   * Default data preparation logic for the entity sheet
-   * @param options - (unused)
-   * @override
-   */
-  getData(options?: Application.RenderOptions): D | Promise<D>;
+    /**
+     * @override
+     */
+    get title(): string;
 
-  /**
-   * @override
-   */
-  protected _getHeaderButtons(): Application.HeaderButton[];
+    /**
+     * @override
+     */
+    close(options?: FormApplication.CloseOptions): Promise<void>;
 
-  /**
-   * @param event - (unused)
-   * @override
-   */
-  protected _updateObject(event: Event, formData: object): Promise<O>;
-}
+    /**
+     * @override
+     */
+    getData(options?: Application.RenderOptions): D | Promise<D>;
 
-declare namespace DocumentSheet {
-  /**
-   * @typeParam D - the type of the data in the Entity
-   * @typeParam O - the type of the Entity which should be managed by this form sheet
-   */
-  interface Data<O extends Entity<any, any> = Entity<any, any>> {
-    cssClass: string;
-    editable: boolean;
-    document: foundry.utils.Duplicated<O['data']>;
-    limited: boolean;
-    options: Options;
-    owner: boolean;
-    title: string;
-  }
+    /**
+     * @override
+     */
+    render(force?: boolean, options?: Application.RenderOptions): this;
 
-  interface Options extends FormApplication.Options {
     /**
-     * @defaultValue `['sheet']`
+     * @override
      */
-    classes: string[];
+    protected _getHeaderButtons(): Application.HeaderButton[];
 
     /**
-     * @defaultValue
-     * ```javascript
-     * `templates/sheets/${this.name.toLowerCase()}.html`
-     * ```
+     * @param event - (unused)
+     * @override
+     */
+    protected _updateObject(event: Event, formData: object): Promise<O>;
+
+    /**
+     * @deprecated since 0.8.0
      */
-    template: string;
+    get entity(): O;
+  }
 
+  namespace DocumentSheet {
     /**
-     * @defaultValue {@link ENTITY_PERMISSIONS.LIMITED}
+     * @typeParam D - the type of the data in the Entity
+     * @typeParam O - the type of the Entity which should be managed by this form sheet
      */
-    viewPermission: foundry.CONST.EntityPermission;
+    interface Data<O extends Document<any, any> = Document<any, any>> {
+      cssClass: string;
+      editable: boolean;
+      document: O;
+      data: ReturnType<O['toObject']>;
+      limited: boolean;
+      options: Options;
+      owner: boolean;
+      title: string;
+    }
```

Comment:
This also contains the deprecated `entity`, which is just a getter for `data`. In our currently used typescript version, we can't have getters on interfaces yet. so just add
```ts
      readonly entity: this['data'];
```